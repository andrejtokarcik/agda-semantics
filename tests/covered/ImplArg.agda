-- https://github.com/bitonic/tog/wiki/Implicit-Arguments
-- aj s popisom checkingu od Andreasa Abela

module ImplArg where
data Nat : Set where
  zero : Nat
  suc  : Nat -> Nat

data Vec (A : Set) : Nat -> Set where
  vnil  : Vec A zero
  vcons : {n : Nat} -> A -> Vec A n -> Vec A (suc n)

Cons = {A : Set} (a : A) {n : Nat} -> Vec A n -> Vec A (suc n)

cons : Cons
cons a = vcons a
