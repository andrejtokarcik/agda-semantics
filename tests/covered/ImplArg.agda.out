<k>
    .K
</k>
<mgu>
    subst(.KList)
</mgu>
<ctx-stack>
    .List
</ctx-stack>
<sig>
    Fun ( "Cons" ) |-> Set ( 1 )
    Fun ( "cons" ) |-> Fun ( "Cons" )
    Data ( "Nat" ) |-> Set ( 0 )
    Data ( "Vec" ) |-> ( #symVariable(7) : Set ( 0 ) ) -> ( #symVariable(6) : 
      Data ( "Nat" ) ) -> Set ( 0 )
    Con ( "suc" ) |-> ( #symVariable(5) : Data ( "Nat" ) ) -> Data ( "Nat" )
    Con ( "vcons" ) |-> { #symVariable(40) : Set ( 0 ) } -> { #symVariable(39)
       : Data ( "Nat" ) } -> ( #symVariable(37) : #symVariable(40) ) -> ( 
      #symVariable(38) : ((Data ( "Vec" ) #symVariable(40)) #symVariable(39)) 
      ) -> ((Data ( "Vec" ) #symVariable(40)) (Con ( "suc" ) #symVariable(39))
      )
    Con ( "vnil" ) |-> { #symVariable(16) : Set ( 0 ) } -> ((Data ( "Vec" ) 
      #symVariable(16)) Con ( "zero" ))
    Con ( "zero" ) |-> Data ( "Nat" )
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "Cons" ) |-> Fun ( "Cons" )
    Name ( "Nat" ) |-> Data ( "Nat" )
    Name ( "Vec" ) |-> Data ( "Vec" )
    Name ( "cons" ) |-> Fun ( "cons" )
    Name ( "suc" ) |-> Con ( "suc" )
    Name ( "vcons" ) |-> Con ( "vcons" )
    Name ( "vnil" ) |-> Con ( "vnil" )
    Name ( "zero" ) |-> Con ( "zero" )
</ncat>
<fun>
    Fun ( "Cons" ) |-> { Var ( "A" ) : Set ( 0 ) } -> ( Var ( "a" ) : Var 
      ( "A" ) ) -> { Var ( "n" ) : Data ( "Nat" ) } -> ( #symVariable(3) : ((
      Data ( "Vec" ) Var ( "A" )) Var ( "n" )) ) -> ((Data ( "Vec" ) Var ( "A"
       )) (Con ( "suc" ) Var ( "n" )))
    Fun ( "cons" ) |-> 'fc((((#symFunClauses(152) { #symVariable(148) }) Var (
       "a" )) { #symVariable(150) }) = (((Con ( "vcons" ) { #symVariable(148) 
      }) { #symVariable(150) }) Var ( "a" )))
</fun>
